### DISCUSSION TITLE
Activate gizmo submit button

### DISCUSSION THREAD

**Discussion by mlandsfeld** [tethysplatform/tethys#795] [Q&A]
Hello,
I am trying to set up a submit button in the left hand panel of my base.html file. I can see it and click on it but "update_maps" request does not appear to be getting sent. Do I need to connect the python controller to that html file?
Thanks!

**Comment by swainn** [comment]
Hi @mlandsfeld there are a few ways that you can use submit buttons to perform actions:

1. Link the submit button to an HTML `<form>` by either including it in the form body or by setting the `form` attribute of the button to be the `id` of the form. This will submit the contents of the form to the URL indicated in the `action` attribute of the form. If no `action` attribute is set, then it will submit to the current URL. See: [HTML Forms - W3Schools](https://www.w3schools.com/html/html_forms.asp) and [The Form element - HTML: HyperText Markup Language | MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form)
2. Alternatively, you can use JavaScript to either submit a form or perform a custom action. To do this, set the `onclick` attribute of the button to a function in your JavaScript or use something like `jQuery.on()` or `jQuery.click()` to bind a function to the onclick event of the button. The function can then contain whatever logic you would like to perform when the button is clicked. See: [HTML onclick Event Attribute | W3Schools](https://www.w3schools.com/tags/ev_onclick.asp), [Element: click event - Web APIs | MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event), [.click() | jQuery API Documentation](https://api.jquery.com/click/), and [.on() | jQuery API Documentation](https://api.jquery.com/on/).
